src/routes/+page.svelte

```diff
@@ -1,14 +1,39 @@
 <script lang="ts">
   import { display } from 'colorjs.io/fn';
+  import { onMount } from 'svelte';
 
   import Colors from '$lib/components/colors/index.svelte';
   import Footer from '$lib/components/Footer.svelte';
   import Header from '$lib/components/Header.svelte';
   import Ratio from '$lib/components/ratio/index.svelte';
-  import { bg, fg } from '$lib/stores';
+  import { bg, fg, format } from '$lib/stores';
+  import { hashToStoreValues, storeValuesToHash } from '$src/lib/utils';
 
   $: bg_fallback = display($bg);
   $: fg_fallback = display($fg);
+
+  onMount(() => {
+    hashToColors();
+    bg.subscribe(colorsToHash);
+    fg.subscribe(colorsToHash);
+    format.subscribe(colorsToHash);
```

Comment:
I realized this is needed after all if we want the hash to update when switching between RGB and Hex (in which case the colors themselves aren't changing, because it's the same space). Re-added in https://github.com/oddbird/oddcontrast/commit/a4486d2f2f2c050d89bfd5c711137b36b269dc88, since the debouncing means there's no harm in calling it more than necessary.